app/src/test/java/com/example/cleanarchitecture/viewmodel/CharacterInformationViewModelTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.cleanarchitecture.viewmodel
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.example.domain.entity.CharacterData
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@ExperimentalCoroutinesApi
+@RunWith(MockitoJUnitRunner::class)
+class CharacterInformationViewModelTest {
+
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var viewModel: CharacterInformationViewModel
+
+    private val characterData: CharacterData = mock()
+    private val successResponse: FragmentData = FragmentData(responseType = FragmentStatus.INITIALIZATION, data = characterData)
+    private val failureResponse: FragmentData =
+        FragmentData(responseType = FragmentStatus.INITIALIZATION_EMPTY_DESCRIPTION, data = characterData)
+
+    @Before
+    fun `set up`() {
+        Dispatchers.setMain(testDispatcher)
+        viewModel = CharacterInformationViewModel()
+    }
+
+    @After
+    fun `after`() {
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
```

Comment:
if you repeat this in each viewmodel test you can move the functionality to a parent class BaseViewModelTest and extend from it

Reply:
done